### DISCUSSION TITLE
Multiple mutations logged in subscribe

### DISCUSSION THREAD

**Discussion by 3zzy** [vuejs/pinia#701] [Help and Questions]
When I subscribe to a store like so:

```
ScopeStore.$subscribe((mutation, state) => {
    console.log({ mutation, state });

    if (mutation.type === 'direct') {
        Emitter.emit('update', { events: mutation.events });
    }
    else if (mutation.type === 'patch object') {
        Emitter.emit('update', { events: mutation.events, payload: mutation.payload });
    }
});
```

... and make a change using patch object, I correctly see it logged but then immediately after there's a "direct" mutation as well which is causing issues:
<img width="907" alt="Screen Shot 2021-10-06 at 6 17 20 am" src="https://user-images.githubusercontent.com/105034/136088844-9fb19838-eb60-491d-b80e-e381592bf7a4.png">

What am I doing wrong? I'm pretty sure I'm not mutating the store anywhere else.

**Comment by posva** [comment]
If you change the state, it's normal for $subscribe to trigger. Is there something not working?

**Reply by 3zzy** [reply]
@posva The screenshot shows two mutations logged one 'direct' and 'patch object'. I'm mutating via patch object so thats expected but the other 'direct' mutation is the issue, its unexpected and causing issues.

**Reply by posva** [reply]
are you sure you aren't changing the store somewhere?
Do you have a boiled down reproduction?

**Reply by 3zzy** [reply]
Yes I double checked and I'm not changing the store anywhere else, tried with a few different values to confirm. 

I came across this similar issue from before https://github.com/posva/pinia/discussions/701 but I don't suppose it would occur in rc6?

**Reply by posva** [reply]
I can't do more without a boiled down repro. You should also update pinia, Rc6 is quite outdated now.

**Reply by 3zzy** [reply]
I'm now using the latest version and tried looking in every file but I still can't resolve this. If I comment out these lines there are no mutations logged so I believe its safe to assume this is the offending code:

```
let fromDate = picker.startDate.format("YYYY-MM-DD 00:00:00"),
    toDate = picker.endDate.format("YYYY-MM-DD 23:59:59"),
    submitted = `${fromDate},${toDate}`;

ScopeStore.$patch((state) => {
    state.fromDate = fromDate;
    state.toDate = toDate;
    state.submitted = submitted;
});
```

Also if you look at the mutations logged it has duplicated in a single mutation itself:
<img width="895" alt="Screen Shot 2021-11-22 at 5 29 05 pm" src="https://user-images.githubusercontent.com/105034/142812550-69453ab3-ac69-4129-aeb5-3653c07763c1.png">

**Reply by 3zzy** [reply]
@posva This action was the culprit:

```
actions: {
    setGlobalScope(obj = {}) {
        this.$patch(obj)
        return this;
    }
}
```

For some reason it logged three separate mutations for separate keys instead of just one.
